<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated with glade 3.20.0 -->
<!-- Edited with love -->
<interface>
  <requires lib="gtk+" version="3.14"/>
  <object class="GtkAdjustment" id="uart_channel_adjustment">
    <property name="upper">100</property>
    <property name="step_increment">1</property>
    <property name="page_increment">10</property>
    <signal name="value-changed" handler="uart_channel_changed" swapped="yes"/>
  </object>
  <object class="GtkAdjustment" id="uart_time_offset_adjustment">
    <property name="upper">9999999</property>
    <property name="step_increment">1</property>
    <property name="page_increment">10</property>
    <signal name="value-changed" handler="uart_time_offset_changed" swapped="yes"/>
  </object>
  <object class="GtkAdjustment" id="udp_channel_adjustment">
    <property name="upper">100</property>
    <property name="step_increment">1</property>
    <property name="page_increment">10</property>
    <signal name="value-changed" handler="udp_channel_changed" swapped="yes"/>
  </object>
  <object class="GtkAdjustment" id="udp_port_adjustment">
    <property name="lower">1024</property>
    <property name="upper">65535</property>
    <property name="step_increment">1</property>
    <property name="page_increment">1</property>
    <signal name="value-changed" handler="udp_port_changed" swapped="yes"/>
  </object>
  <object class="GtkAdjustment" id="udp_time_offset_adjustment">
    <property name="upper">9999999</property>
    <property name="step_increment">1</property>
    <property name="page_increment">10</property>
    <signal name="value-changed" handler="udp_time_offset_changed" swapped="yes"/>
  </object>
  <object class="GtkAdjustment" id="virtual_channel_adjustment">
    <property name="upper">100</property>
    <property name="step_increment">1</property>
    <property name="page_increment">10</property>
    <signal name="value-changed" handler="virtual_channel_changed" swapped="yes"/>
  </object>
  <object class="GtkAdjustment" id="virtual_time_offset_adjustment">
    <property name="upper">9999999</property>
    <property name="step_increment">1</property>
    <property name="page_increment">10</property>
    <signal name="value-changed" handler="virtual_time_offset_changed" swapped="yes"/>
  </object>
  <object class="GtkGrid" id="sensor_control_view">
    <property name="visible">True</property>
    <property name="can_focus">False</property>
    <property name="row_spacing">4</property>
    <property name="column_spacing">4</property>
    <property name="column_homogeneous">True</property>
    <child>
      <object class="GtkLabel" id="port_label">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <property name="halign">end</property>
        <property name="label" translatable="yes">Port name</property>
      </object>
      <packing>
        <property name="left_attach">0</property>
        <property name="top_attach">0</property>
      </packing>
    </child>
    <child>
      <object class="GtkComboBoxText" id="ports_cbt">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <signal name="changed" handler="port_changed" swapped="yes"/>
      </object>
      <packing>
        <property name="left_attach">1</property>
        <property name="top_attach">0</property>
        <property name="width">2</property>
      </packing>
    </child>
    <child>
      <object class="GtkLabel" id="port_type_title_label">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <property name="halign">end</property>
        <property name="label" translatable="yes">Port type</property>
      </object>
      <packing>
        <property name="left_attach">0</property>
        <property name="top_attach">1</property>
      </packing>
    </child>
    <child>
      <object class="GtkLabel" id="port_type_name_label">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <property name="halign">start</property>
        <property name="label" translatable="yes">n/a</property>
      </object>
      <packing>
        <property name="left_attach">1</property>
        <property name="top_attach">1</property>
        <property name="width">2</property>
      </packing>
    </child>
    <child>
      <object class="GtkStack" id="params_stack">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <property name="homogeneous">False</property>
        <child>
          <object class="GtkGrid" id="virtual_port_grid">
            <property name="visible">True</property>
            <property name="can_focus">False</property>
            <property name="row_spacing">4</property>
            <property name="column_spacing">4</property>
            <property name="column_homogeneous">True</property>
            <child>
              <object class="GtkLabel" id="virtual_channel_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Channel</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">0</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="virtual_time_offset_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Time offset</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">1</property>
              </packing>
            </child>
            <child>
              <object class="GtkSpinButton" id="virtual_channel_spin">
                <property name="visible">True</property>
                <property name="can_focus">True</property>
                <property name="adjustment">virtual_channel_adjustment</property>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">0</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkSpinButton" id="virtual_time_offset_spin">
                <property name="visible">True</property>
                <property name="can_focus">True</property>
                <property name="adjustment">virtual_time_offset_adjustment</property>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">1</property>
                <property name="width">2</property>
              </packing>
            </child>
          </object>
          <packing>
            <property name="name">page0</property>
            <property name="title" translatable="yes">page0</property>
          </packing>
        </child>
        <child>
          <object class="GtkGrid" id="uart_port_grid">
            <property name="visible">True</property>
            <property name="can_focus">False</property>
            <property name="row_spacing">4</property>
            <property name="column_spacing">4</property>
            <property name="column_homogeneous">True</property>
            <child>
              <object class="GtkLabel" id="uart_channel_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Channel</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">0</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="uart_time_offset_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Time offset</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">1</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="uart_protocol_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Protocol</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="uart_device_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Device</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">3</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="uart_mode_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Mode</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">4</property>
              </packing>
            </child>
            <child>
              <object class="GtkComboBoxText" id="uart_protocol_cbt">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <signal name="changed" handler="uart_protocol_changed" swapped="yes"/>
                <items>
                  <item id="SAS" translatable="yes">SAS</item>
                  <item id="NMEA" translatable="yes">NMEA</item>
                </items>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">2</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkSpinButton" id="uart_channel_spin">
                <property name="visible">True</property>
                <property name="can_focus">True</property>
                <property name="adjustment">uart_channel_adjustment</property>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">0</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkSpinButton" id="uart_time_offset_spin">
                <property name="visible">True</property>
                <property name="can_focus">True</property>
                <property name="adjustment">uart_time_offset_adjustment</property>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">1</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkComboBoxText" id="uart_device_cbt">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <signal name="changed" handler="uart_device_changed" swapped="yes"/>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">3</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkComboBoxText" id="uart_mode_cbt">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <signal name="changed" handler="uart_mode_changed" swapped="yes"/>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">4</property>
                <property name="width">2</property>
              </packing>
            </child>
          </object>
          <packing>
            <property name="name">page1</property>
            <property name="title" translatable="yes">page1</property>
            <property name="position">1</property>
          </packing>
        </child>
        <child>
          <object class="GtkGrid" id="udp_port_grid">
            <property name="visible">True</property>
            <property name="can_focus">False</property>
            <property name="row_spacing">4</property>
            <property name="column_spacing">4</property>
            <property name="column_homogeneous">True</property>
            <child>
              <object class="GtkComboBoxText" id="udp_protocol_cbt">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <signal name="changed" handler="udp_protocol_changed" swapped="yes"/>
                <items>
                  <item id="SAS" translatable="yes">SAS</item>
                  <item id="NMEA" translatable="yes">NMEA</item>
                </items>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">2</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkSpinButton" id="udp_channel_spin">
                <property name="visible">True</property>
                <property name="can_focus">True</property>
                <property name="adjustment">udp_channel_adjustment</property>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">0</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkSpinButton" id="udp_time_offset_spin">
                <property name="visible">True</property>
                <property name="can_focus">True</property>
                <property name="adjustment">udp_time_offset_adjustment</property>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">1</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkSpinButton" id="udp_port_spin">
                <property name="visible">True</property>
                <property name="can_focus">True</property>
                <property name="adjustment">udp_port_adjustment</property>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">4</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkComboBoxText" id="udp_addr_cbt">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <signal name="changed" handler="udp_addr_changed" swapped="yes"/>
              </object>
              <packing>
                <property name="left_attach">1</property>
                <property name="top_attach">3</property>
                <property name="width">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="udp_channel_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Channel</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">0</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="udp_time_offset_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Time offset</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">1</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="udp_protocol_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">Protocol</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">2</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="udp_addr_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">IP address</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">3</property>
              </packing>
            </child>
            <child>
              <object class="GtkLabel" id="udp_port_number_label">
                <property name="visible">True</property>
                <property name="can_focus">False</property>
                <property name="halign">end</property>
                <property name="label" translatable="yes">UDP port</property>
              </object>
              <packing>
                <property name="left_attach">0</property>
                <property name="top_attach">4</property>
              </packing>
            </child>
          </object>
          <packing>
            <property name="name">page2</property>
            <property name="title" translatable="yes">page2</property>
            <property name="position">2</property>
          </packing>
        </child>
      </object>
      <packing>
        <property name="left_attach">0</property>
        <property name="top_attach">4</property>
        <property name="width">3</property>
      </packing>
    </child>
    <child>
      <object class="GtkSeparator">
        <property name="visible">True</property>
        <property name="can_focus">False</property>
        <property name="margin_top">4</property>
        <property name="margin_bottom">4</property>
      </object>
      <packing>
        <property name="left_attach">0</property>
        <property name="top_attach">3</property>
        <property name="width">3</property>
      </packing>
    </child>
    <child>
      <object class="GtkCheckButton" id="enable_checkbtn">
        <property name="label" translatable="yes">Enabled</property>
        <property name="visible">True</property>
        <property name="can_focus">True</property>
        <property name="receives_default">False</property>
        <property name="draw_indicator">True</property>
        <signal name="toggled" handler="enable_toggled" swapped="yes"/>
      </object>
      <packing>
        <property name="left_attach">1</property>
        <property name="top_attach">2</property>
        <property name="width">2</property>
      </packing>
    </child>
  </object>
  <object class="GtkSizeGroup">
    <property name="mode">vertical</property>
    <widgets>
      <widget name="port_label"/>
      <widget name="ports_cbt"/>
      <widget name="port_type_title_label"/>
      <widget name="port_type_name_label"/>
      <widget name="virtual_channel_label"/>
      <widget name="virtual_time_offset_label"/>
      <widget name="virtual_channel_spin"/>
      <widget name="virtual_time_offset_spin"/>
      <widget name="uart_channel_label"/>
      <widget name="uart_time_offset_label"/>
      <widget name="uart_protocol_label"/>
      <widget name="uart_device_label"/>
      <widget name="uart_mode_label"/>
      <widget name="uart_protocol_cbt"/>
      <widget name="uart_channel_spin"/>
      <widget name="uart_time_offset_spin"/>
      <widget name="uart_device_cbt"/>
      <widget name="uart_mode_cbt"/>
      <widget name="udp_protocol_cbt"/>
      <widget name="udp_channel_spin"/>
      <widget name="udp_time_offset_spin"/>
      <widget name="udp_port_spin"/>
      <widget name="udp_addr_cbt"/>
      <widget name="udp_channel_label"/>
      <widget name="udp_time_offset_label"/>
      <widget name="udp_protocol_label"/>
      <widget name="udp_addr_label"/>
      <widget name="udp_port_number_label"/>
      <widget name="enable_checkbtn"/>
    </widgets>
  </object>
</interface>
